pkg/apply/applier.go

```diff
@@ -334,17 +329,21 @@ func (a *Applier) Run(ctx context.Context) <-chan event.Event {
 			UseCache:     true,
 		})
 		if err != nil {
-			eventChannel <- event.Event{
-				Type: event.ErrorType,
-				ErrorEvent: event.ErrorEvent{
-					Err: err,
-				},
-			}
+			handleError(eventChannel, err)
 		}
 	}()
 	return eventChannel
 }
 
+func handleError(eventChannel chan event.Event, err error) {
```

Comment:
Maybe we can create a common or util package to house these common routines in a future PR. Looks like the same function is in the polling engine.